posts/python-in-production-part-4-of-5.md

```diff
@@ -0,0 +1,288 @@
+
+# Create a Unit File
+
+Systemd unit files live on the file system of your linux server.  A commonly used place for *user application unit files* is `/etc/systemd/user/{file_name}.service`.  The `{file_name}` is usually the name of your program.  For our application, the full path plus the file name will be `/etc/systemd/user/simple_api.service`.  For now, we can create the file here in our local repo, and then *transfer* it to our linux server once we are finished editing it.
+
+You should be in the top level directory of your project.  For us, that is `sample-python-program/`.  With your terminal, create the unit file.
+
+```bash
+touch simple_api.service
+```
+
+Now with your text editor, write the following code to the file.  Don't worry, we will review what each line means right after we save the file.
+
+```ini
+[Unit]
+Description=Simple API Service
+After=network.target
+
+[Service]
+ExecStart=/usr/local/bin/simple_api
+Restart=always
+
+[Install]
+WantedBy=multi-user.target
+```
+
+Following is an explanation of each line.
+* `[Unit]` This section describes your service in the context of the rest of the system.
+* `Description=Simple API Service` is a short description of the service.
+* `After=network.target` tells systemd to wait until the network is up before starting our service.  This makes sense for most web applications.
+* `[Service]` This section handles our application specific configuration.
+* `ExecStart=/usr/local/bin/simple_api` tells systemd where your program binary lives.  We will need to transfer our binary to this location.
+* `Restart=always` tells systemd to restart our program if it closes for any reason.
+* `[Install]` This section handles your service in the context of the user space.
+* `WantedBy=multi-user.target` tells systemd to make our program available to multiple users on the linux system.
+
+Now that we have our systemd unit file, we are ready to put it, along with our program binary, onto our linux server.
+
+# Putting it on the Server
+
+**Note:** This section involves some pretty advanced Linux topics.  It's totally cool if you don't fully understand everything.  I will offer some minor explanations, but I highly encourage you to read up on any and all commands that you run on your machines.  Please don't *blindly run* linux commands that you find on the internet.
+
+If you have been following along since the beginning of the series, then you will have a *binary* file called `simple_api` and a *systemd unit file* called `simple_api.service` in your current working directory.  In order to get these files onto our linux server, we are going to use a utility called ***scp***.
+
+SCP stands for **S**ecure **C**opy.  It is a simple command line utility that transfers files from your local machine, to a remote server (or the other way around!).  If you are more comfortable with an ftp utility, or a program such as filezilla, feel free to use that instead.  
```

Comment:
You've only highlighted the S & C of SCP. Isn't SCP, Secure Copy Protocol?

Reply:
You might also want to highlight what makes the transfer secure specifically here versus just running a cp